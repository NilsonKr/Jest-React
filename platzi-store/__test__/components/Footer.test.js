import React from 'react';
import Footer from '../../src/components/Footer';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import { expect } from '@jest/globals';

describe('<Footer />', () => {
  const wrapper = shallow(<Footer />);

  test('Should Render ', () => {
    expect(wrapper.length).toBe(1);
  });

  test('There is right Info', () => {
    expect(wrapper.find('p').length).toBeGreaterThanOrEqual(2);
  });

  test('Right Title', () => {
    expect(wrapper.find('.Footer-title').text()).toBe('Platzi Store');
  });

  test('Footer Snapshot', () => {
    const footerSnap = create(<Footer />);

    expect(footerSnap.toJSON()).toMatchSnapshot();
  });
});
