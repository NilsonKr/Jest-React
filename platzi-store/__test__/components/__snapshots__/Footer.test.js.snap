// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer Snapshot 1`] = `
<div
  className="Footer"
>
  <p
    className="Footer-title"
  >
    Platzi Store
  </p>
  <p
    className="Footer-copy"
  >
    Todos los Izquierdos Reservados
  </p>
</div>
`;
